spirv_hlsl.cpp

```diff
@@ -4412,6 +4420,51 @@ void CompilerHLSL::emit_access_chain(const Instruction &instruction)
 	}
 }
 
+bool CompilerHLSL::should_emit_temporary(uint32_t base, const SPIRType &target_type) const
+{
+	return flattened_buffer_blocks.count(base) && target_type.basetype == SPIRType::Struct;
+}
+
+std::string CompilerHLSL::flattened_access_chain_struct(uint32_t base, const uint32_t *indices, uint32_t count,
+                                                        const SPIRType &target_type, uint32_t offset)
+{
+	std::string expr;
+
+	expr += "{";
+
+	for (uint32_t i = 0; i < uint32_t(target_type.member_types.size()); ++i)
```

Comment:
It might be possible to just modify CompilerGLSL implementation with a check to backend.can_declare_struct_inline. This seems like duplicated code.